### PR TITLE
[breaking change] use common data structure for all annotations

### PR DESCRIPTION
**Pull request by smolkaj**
This is non-backward-compatible solution to issue #285. To be included in 2.0.

### CODE REVIEW

**Comment by antoninbas** [comment]
@smolkaj this should probably be an issue at this stage and not a PR against master, which we cannot merge for a while.

It is possible that time has come for us to create a development branch for P4Runtime v2.0, but even then it makes sense to have a corresponding issue that we can use for tracking.

**Comment by jafingerhut** [comment]
It is entirely up to the working group how they want to handle this in the Github repo files, but an alternative to creating a branch is to have separate directories for v1 and v2 of the P4Runtime API spec, and initially populate v2 files with copies of the latest v1 files.

There are disadvantages to such an approach, but a potential advantage is to make it easier to continue to make small corrections/clarifications to the v1 files as time goes on, and easily keep them "in the file system" for operations like auto-generating HTML/PDF from both v1 and v2 without doing git checkout operations.

**Comment by smolkaj** [comment]
I created #288 to track this.

**Comment by smolkaj** [comment]
I will close this PR until it becomes relevant again.

File: proto/p4/config/v1/p4types.proto [hunk 1]
```diff
@@ -189,11 +201,11 @@ message P4EnumTypeSpec {
   message Member {
     string name = 1;
     repeated string annotations = 2;
```
Comment (chrispsommers) [inline proto/p4/config/v1/p4types.proto:203]: Why isn't this deleted? (same comment for other occurrences of `string annotations`

Comment (smolkaj) [reply proto/p4/config/v1/p4types.proto]: Good catch, fixed :)

**Comment by smolkaj** [comment]
Just opening this temporarily to incorporate updates pointed out by @chrispsommers .

**Pull request closed without merging**